lib/product-import.coffee

```diff
@@ -161,39 +172,48 @@ class ProductImport
         if existingProduct?
           @_fetchSameForAllAttributesOfProductType(prodToProcess.productType)
           .then (sameForAllAttributes) =>
-            @_validateEnums(prodToProcess, @_cache.productType[prodToProcess.productType.id])
-            .then (updateActions) =>
-              if updateActions and _.size(updateActions.actions) > 0 then @_updateEnumUpdateActions(enumUpdateActions, updateActions)
-              @_prepareUpdateProduct(prodToProcess, existingProduct).then (preparedProduct) =>
-                synced = @sync.buildActions(preparedProduct, existingProduct, sameForAllAttributes)
-                if synced.shouldUpdate()
-                  @client.products.byId(synced.getUpdateId()).update(synced.getUpdatePayload())
-                else
-                  Promise.resolve statusCode: 304
+            @_prepareUpdateProduct(prodToProcess, existingProduct).then (preparedProduct) =>
+              synced = @sync.buildActions(preparedProduct, existingProduct, sameForAllAttributes)
+              if synced.shouldUpdate()
+                @client.products.byId(synced.getUpdateId()).update(synced.getUpdatePayload())
+              else
+                Promise.resolve statusCode: 304
         else
           @_prepareNewProduct(prodToProcess)
           .then (product) =>
-            @_validateEnums(product, @_cache.productType[product.productType.id])
-            .then (updateActions) =>
-              if updateActions and _.size(updateActions) > 0 then @_updateEnumUpdateActions(enumUpdateActions, updateActions)
-              @client.products.create(product)
+            @client.products.create(product)
 
-    @_updateProductType(enumUpdateActions)
-    .then ->
-      debug 'About to send %s requests', _.size(posts)
-      Promise.settle(posts)
+    debug 'About to send %s requests', _.size(posts)
+    Promise.settle(posts)
 
   _filterAttributes: (product) =>
     new Promise (resolve) =>
       if @filterUnknownAttributes
-        @_ensureProductTypeInMemory(product.productType.id)
-        .then =>
-          @unknownAttributesFilter.filter(@_cache.productType[product.productType.id],product)
+        @unknownAttributesFilter.filter(@_cache.productType[product.productType.id],product)
         .then (filteredProduct) ->
           resolve(filteredProduct)
       else
         resolve(product)
 
+
+  # updateActions are of the form:
+  # { productTypeId: [{updateAction},{updateAction},...],
+  #   productTypeId: [{updateAction},{updateAction},...]
+  # }
+  _filterUniqueUpdateActions: (updateActions) =>
+    uniqueUpdateActions = {}
+    _.each _.keys(updateActions), (productTypeId) =>
```

Comment:
tired... was not able to get the output I need. Are u sure I need it here?
have a look at the test for what I am actually doing here: https://github.com/sphereio/sphere-product-import/commit/7a864f2443e526d6f6b5dc50ed70ce483a4df1e8